### PR TITLE
[Storage] Update truncation tool to support recent state kv db change.

### PR DESCRIPTION
**Pull request by grao1991**
### Description
Share the same code to sync the databases for production and truncation tool.
Also fix a bug that we didn't write the progress during truncation.

### Test Plan
UT

### CODE REVIEW

**Review by zekun000 — APPROVED** [review]

File: storage/aptosdb/src/state_store/mod.rs [hunk 1]
```diff
@@ -284,6 +284,44 @@ impl StateStore {
...
+            .expect("buffered state creation failed."),
+        );
+        Self {
+            state_db,
+            buffered_state,
+            buffered_state_target_items,
+        }
+    }
+
+    pub fn sync_commit_progress(
```
Comment (areshand) [inline storage/aptosdb/src/state_store/mod.rs:320]: nit: would be helpful to add some comments on why truncating ledger and state_kv db to overall progress to keep them in sync.

**Review by areshand — APPROVED** [review]

File: storage/aptosdb/src/db_debugger/truncate/mod.rs [hunk 2]
```diff
@@ -103,14 +118,18 @@ impl Cmd {
...
-            ledger_db_version,
-            self.target_version,
-            self.ledger_db_batch_size,
+        println!("Starting ledger db and state kv db truncation...");
+        let batch = SchemaBatch::new();
+        batch.put::<DbMetadataSchema>(
+            &DbMetadataKey::OverallCommitProgress,
+            &DbMetadataValue::Version(self.target_version),
         )?;
+        ledger_db.write_schemas(batch)?;
```
Comment (msmouse) [inline storage/aptosdb/src/db_debugger/truncate/mod.rs:127]: looks like there's no drawback to use `DB::put()` directly as we discussed :P

**Review by msmouse — APPROVED** [review]

File: storage/aptosdb/src/state_store/mod.rs (hunk 1 above)
Comment (grao1991) [reply storage/aptosdb/src/state_store/mod.rs:320]: Done.

File: storage/aptosdb/src/db_debugger/truncate/mod.rs (hunk 2 above)
Comment (grao1991) [reply storage/aptosdb/src/db_debugger/truncate/mod.rs:127]: Done.

File: storage/schemadb/src/lib.rs [hunk 3]
```diff
@@ -191,7 +191,6 @@ impl DB {
     /// Writes single record.
     pub fn put<S: Schema>(&self, key: &S::Key, value: &S::Value) -> Result<()> {
         // Not necessary to use a batch, but we'd like a central place to bump counters.
-        // Used in tests only anyway.
```
Comment (msmouse) [inline storage/schemadb/src/lib.rs:194]: ❤️

**Pull request merged**